src/odemis/acq/move.py

```diff
@@ -767,14 +845,103 @@ def _doCryoSwitchSamplePosition(future, target):
             # run the moves
             logging.info("Moving from position {} to position {}.".format(current_name, target_name))
             for component, sub_move in sub_moves:
-                run_sub_move(future, component, sub_move)
+                self._run_sub_move(future, component, sub_move)
+
+        except CancelledError:
+            logging.info("CryoSwitchSamplePosition cancelled.")
+        except Exception:
+            logging.exception("Failure to move to {} position.".format(target_name))
+            raise
+        finally:
+            with future._task_lock:
+                if future._task_state == CANCELLED:
+                    raise CancelledError()
+                future._task_state = FINISHED
+
+
+class MimasPostureManager(MicroscopePostureManager):
+    def __init__(self, microscope):
+        # Load components
+        self.stage = model.getComponent(role="stage")
+        self.align = model.getComponent(role="align")
+        self.gis = model.getComponent(role="gis")
+        # set linear axes and rotational axes used
+        self.axes = self.stage.axes
+        self.linear_axes = set(key for key in self.axes.keys() if key in {'x', 'y', 'z', 'm'})
+        self.rotational_axes = set(key for key in self.axes.keys() if key in {'rx', 'ry', 'rz', 'rm'})
```

Comment:
Please add a few check for the metadata. At least, from the code it's clear these 3 metadata are required: MD_FAV_POS_ACTIVE, MD_FAV_POS_DEACTIVE, MD_POS_ACTIVE_RANGE.
The aligner also has some metadata requirements.